### ISSUE TITLE
Mapping plugin versions to Snap daemon versions

### ISSUE THREAD

**Issue by andrzej-k** [intelsdi-x/snap#1403]
**Problem statement:**
Due to constantly growing number of plugins (and their versions) and the fact that we reached 1.0 release, I'd propose to provide a mechanism which would allow Snap users to easily identify with which versions of the Snap framework given plugin works. Trying to load a plugin to discover that it doesn't work with given version of Snap framework can be bad experience. 
Consider this example of new releases for Snap framework and Snap plugin:

|         Snap version        |         Plugin version        |
|:------------------:|:------------------:|
| snap v.1           | plugin v.3 <br> supported by: snap v.1   |
| snap v.2           | plugin v.3 <br> supported by: snap v.1, v.2  |
|                    | plugin v.4 <br> supported by: snap v.1, v.2   |
| snap v.3 <br> breaking change in plugin's interface | plugin v.5 <br> supported by: snap v.3  |

**Some requirements:**
- Each version of plugin should clearly indicate with which version of framework it works. This is also to allow Users who are not migrating to the newest version of framework every time one is released to find a plugin which works with their version of framework.

- Support matrix information should be easily consumable by humans and machines. Meaning that if User has an automation framework for loading plugins then this framework should be able to get support matrix information easily - so this information must have standard format.

**Implementation:**
My first thought was to leverage ```metadata.yml``` and then use it to auto populate support matrix in plugin's README. Including ```metadata.yml``` as part of release package could allow automation tools to discover this information easily. 
Plugin's help message should also indicate supported framework versions - could possibly also leverage ```metadata.yml``` for this.

**Future:**
Having that kind of information, possibly also merged with information which metrics are exposed by given version of plugin, would greatly help operating Snap and automating plugin loading and task starting activities.

**Comment by sandlbn** [comment]
@andrzej-k Great idea, but this will add more work for developers. I think we should do this by adding this as an step in the large test. Large test can be executed against multiple versions of snap daemons. User should also have visibility which version of the snapd is supported and which is not. Right now we have too many snapteld versions.

**Comment by andrzej-k** [comment]
@sandlbn I totally agree - this has to be automated, any manual updates may be incorrect, not in time, etc.

**Comment by nanliu** [comment]
As we slowly roll out large tests, our current matrix is simply last release and current snap build. It would be awesome to start expressing the versions as a range in each plugin release:

```
requirements:
  snap:
    version: ">= 1.0.0 < 2.0.0"
  psutil:
    version: '...'
```
